types/dataset.d.ts

```diff
@@ -0,0 +1,379 @@
+ * Each dataset item is stored as a separate JSON file, where `{INDEX}` is a zero-based index of the item in the dataset.
+ *
+ * If the `APIFY_TOKEN` environment variable is set but `APIFY_LOCAL_STORAGE_DIR` not, the data is stored in the
+ * <a href="https://docs.apify.com/storage/dataset" target="_blank">Apify Dataset</a>
+ * cloud storage. Note that you can force usage of the cloud storage also by passing the `forceCloud`
+ * option to [`Apify.openDataset()`](apify#module_Apify.openDataset) function,
+ * even if the `APIFY_LOCAL_STORAGE_DIR` variable is set.
+ *
+ * **Example usage:**
+ *
+ * ```javascript
+ * // Write a single row to the default dataset
+ * await Apify.pushData({ col1: 123, col2: 'val2' });
+ *
+ * // Open a named dataset
+ * const dataset = await Apify.openDataset('some-name');
+ *
+ * // Write a single row
+ * await dataset.pushData({ foo: 'bar' });
+ *
+ * // Write multiple rows
+ * await dataset.pushData([
+ *   { foo: 'bar2', col2: 'val2' },
+ *   { col3: 123 },
+ * ]);
+ * ```
+ * @hideconstructor
+ */
+export class Dataset {
+    constructor(datasetId: string, datasetName: string);
+    datasetId: string;
+    datasetName: string;
+    /**
+     * Stores an object or an array of objects to the dataset.
+     * The function returns a promise that resolves when the operation finishes.
+     * It has no result, but throws on invalid args or other errors.
+     *
+     * **IMPORTANT**: Make sure to use the `await` keyword when calling `pushData()`,
+     * otherwise the actor process might finish before the data is stored!
+     *
+     * The size of the data is limited by the receiving API and therefore `pushData()` will only
+     * allow objects whose JSON representation is smaller than 9MB. When an array is passed,
+     * none of the included objects
+     * may be larger than 9MB, but the array itself may be of any size.
+     *
+     * The function internally
+     * chunks the array into separate items and pushes them sequentially.
+     * The chunking process is stable (keeps order of data), but it does not provide a transaction
+     * safety mechanism. Therefore, in the event of an uploading error (after several automatic retries),
+     * the function's Promise will reject and the dataset will be left in a state where some of
+     * the items have already been saved to the dataset while other items from the source array were not.
+     * To overcome this limitation, the developer may, for example, read the last item saved in the dataset
+     * and re-attempt the save of the data from this item onwards to prevent duplicates.
+     *
+     * @param {Object|Array} data Object or array of objects containing data to be stored in the default dataset.
+     * The objects must be serializable to JSON and the JSON representation of each object must be smaller than 9MB.
+     * @return {Promise<void>}
+     */
+    pushData(data: any): Promise<void>;
```

Comment:
`pushData<T>(data: T)` if the Dataset class won't be defined as generic, but would be more useful if the class itself was defined as generic, that fallsback to `<T = any>`

Reply:
I will try that.

Reply:
Again, please write this down and come back to it later.